code/modules/robotics/bot/barbot.dm

```diff
@@ -0,0 +1,199 @@
+							"boorbon","beepskybeer","screwdriver","bloody_mary","bloody_scary",\
+							"snakebite","diesel","suicider","port","gin","vermouth","bitters","whiskey_sour",\
+							"daiquiri","martini","v_martini","murdini","manhattan","libre","ginfizz","gimlet",\
+							"v_gimlet","w_russian","b_russian","irishcoffee","cosmo","beach","gtonic","vtonic","sonic",\
+							"gpink","eraser","squeeze","hunchback","madmen","planter","maitai","harlow",\
+							"gchronic","margarita","tequini","pfire","bull","longisland","pinacolada","longbeach",\
+							"mimosa","french75","sangria","tomcollins","peachschnapps","moscowmule","tequilasunrise",\
+							"paloma","mintjulep","mojito","cremedementhe","grasshopper","curacao","bluelagoon",\
+							"bluehawaiian","negroni","necroni", "cola", "juice_lime", "juice_lemon", "juice_orange", \
+							"juice_cran", "juice_cherry", "juice_pineapple", "juice_tomato", \
+							"coconut_milk", "sugar", "water", "vanilla", "tea","mint")
+
+	var/possible_poisons = list("acetaldehyde","wolfsbane","ants","weedkiller","cyanide","krokodil", "fluorine", "radium", \
+							"neurotoxin","phlogiston","ghostchilijuice", "cholesterol", "mercury")
+
+	var/possible_vessels = list(/obj/item/reagent_containers/food/drinks/drinkingglass, \
+							/obj/item/reagent_containers/food/drinks/drinkingglass/shot, \
+							/obj/item/reagent_containers/food/drinks/drinkingglass/oldf, \
+							/obj/item/reagent_containers/food/drinks/drinkingglass/round, \
+							/obj/item/reagent_containers/food/drinks/drinkingglass/wine, \
+							/obj/item/reagent_containers/food/drinks/drinkingglass/cocktail, \
+							/obj/item/reagent_containers/food/drinks/drinkingglass/flute)
+
+	var/possible_stuffs = list(/obj/item/cocktail_stuff/drink_umbrella, \
+							/obj/item/cocktail_stuff/maraschino_cherry, \
+							/obj/item/cocktail_stuff/cocktail_olive, \
+							/obj/item/cocktail_stuff/celery)
+
+	var/possible_wedges = list(/obj/item/reagent_containers/food/snacks/plant/orange/wedge, \
+	                        /obj/item/reagent_containers/food/snacks/plant/lime/wedge, \
+							/obj/item/reagent_containers/food/snacks/plant/lemon/wedge, \
+							/obj/item/reagent_containers/food/snacks/plant/grapefruit/wedge)
+
+	New()
+		..()
+		src.setEmotion("happy")
+		// Start by getting a few initial things
+		if (!src.home || !isturf(src.home))
+			src.home = get_turf(src)
+		if (!length(src.homeTables))
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables += T
+
+	proc/setEmotion(var/set_emotion)
+		if(src.emotion == set_emotion)
+			return
+		src.emotion = set_emotion
+		var/emotion_image = image(src.icon, "face-[src.emotion]")
+		src.UpdateOverlays(emotion_image, "emotion_image")
+
+	proc/get_empty_tables()
+		if (!length(src.homeTables))
+			for (var/obj/table/reinforced/bar/T in view(5, src.home))
+				src.homeTables += T
+			if (length(src.homeTables))
+				explode()
```

Comment:
I assume you mean to be checking for the opposite rather than exploding your bot after is successful

Reply:
```suggestion
			if (!length(src.homeTables))
				explode()
```